src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -6495,30 +6383,27 @@ namespace HeatBalanceSurfaceManager {
                                                       QRadSWInAbs(SurfNum) - CTFConstInPart(SurfNum) - construct.CTFCross(0) * TH11) /
                                                      (HMovInsul);
                             // if any mixed heat transfer models in zone, apply limits to CTF result
-                            if (zone_has_mixed_HT_models(ZoneNum))
+                            if (DataHeatBalSurface::Zone_has_mixed_HT_models[ZoneNum])
                                 TempSurfInTmp(SurfNum) = max(
                                     MinSurfaceTempLimit,
                                     min(MaxSurfaceTempLimit, TempSurfInTmp(SurfNum))); // Limit Check //Tuned Precomputed condition to eliminate loop
                         }
 
                     } else { // Window
 
-                        if (construct.SourceSinkPresent && calcHeatBalanceInsideSurfFirstTime)
-                            ShowSevereError("Windows are not allowed to have embedded sources/sinks");
-
                         if (SurfaceWindow(SurfNum).OriginalClass == SurfaceClass_TDD_Diffuser) { // Tubular daylighting device
                             // Lookup up the TDD:DOME object
-                            PipeNum = FindTDDPipe(SurfNum);
-                            SurfNum2 = TDDPipe(PipeNum).Dome;
-                            Ueff = 1.0 / TDDPipe(PipeNum).Reff;
+                            int pipeNum = SurfaceWindow(SurfNum).TDDPipeNum;
+                            int domeNum = DataDaylightingDevices::TDDPipe(pipeNum).Dome;
```

Comment:
The compiler is probably already doing it, but I'd assign `const` where possible to be clear this is just assigned.

Reply:
OK - will do.